lib/ferto/client.rb

```diff
@@ -153,6 +157,21 @@ def build_body(aggr_id, aggr_limit, url, callback_url, callback_type,
         body[:extra] = extra.is_a?(Hash) ? extra.to_json : extra.to_s
       end
 
+      # We will continue to expose the user_agent field just like tools
+      # like curl and wget do. Along with that we will follow their paradigm
+      # where if both a user-agent flag and a `User-Agent` in the request headers
+      # are provided then the user agent in the request headers is preferred.
+      #
+      # Also if the `user_agent` is provided but the request headers do not
+      # contain a `User-Agent` key, then the `user_agent` is copied to the headers
+      if request_headers.is_a?(Hash)
```

Comment:
I'd drop this type check (too much Go?) :)

Reply:
Haha! Maybe! But i mainly did it for consistency, even though the field is initialized as an empty hash. Let's keep it, the code needs to be guarded well for that since we do not want to boom in case we need to update a value ( e.g "User-Agent"). An alternative would be to drop the check but add the following:

```
if user_agent && (request&.respond_to?(:key?) && !request_headers&.key?("User-Agent"))
  request_headers["User-Agent"] = user_agent
end
```

Do you think the latter is better?